code/work_1.sh

```diff
@@ -0,0 +1,170 @@
+#!bin/bash
```

Comment:
shellcheck结果：
In ../2015-linux-public-icoy/code/work_1.sh line 6:
    	for i in $(find $2 -regex  '.*\.jpg');do
                 ^-- SC2044: For loops over find output are fragile. Use find -exec or a while read loop.
                        ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 9:
		filename=$(basename $i)
                                    ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 12:
        	convert $i -quality $1 $2/"$name""%$1".$extension;
                        ^-- SC2086: Double quote to prevent globbing and word splitting.
                                    ^-- SC2086: Double quote to prevent globbing and word splitting.
                                       ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                       ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 21:
	for i in $(find $2 -regex  '.*\.jpg\|.*\.svg\|.*\.png');do
                 ^-- SC2044: For loops over find output are fragile. Use find -exec or a while read loop.
                        ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 23:
		filename=$(basename $i)
                                    ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 26:
		convert $i -resize $1 $2/"$name""--$1".$extension;
                        ^-- SC2086: Double quote to prevent globbing and word splitting.
                                   ^-- SC2086: Double quote to prevent globbing and word splitting.
                                      ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                       ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 34:
	for i in $(find $2 -regex  '.*\.jpg\|.*\.svg\|.*\.png');do
                 ^-- SC2044: For loops over find output are fragile. Use find -exec or a while read loop.
                        ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 36:
		width=$(identify -format %w $i)
                                            ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 38:
		filename=$(basename $i)
                                    ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 43:
		-size "${width}"x30 caption:$1 $i +swap -gravity south \
                                            ^-- SC2086: Double quote to prevent globbing and word splitting.
                                               ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 44:
		-composite $2/"$name""_embed".$extension
                           ^-- SC2086: Double quote to prevent globbing and word splitting.
                                              ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 55:
	for i in $(find $2 -regex  '.*\.*'); do
                 ^-- SC2044: For loops over find output are fragile. Use find -exec or a while read loop.
                        ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 57:
		filename=$(basename $i)
                                    ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 60:
		cp $i $2/$1$name.$extension
                   ^-- SC2086: Double quote to prevent globbing and word splitting.
                      ^-- SC2086: Double quote to prevent globbing and word splitting.
                         ^-- SC2086: Double quote to prevent globbing and word splitting.
                           ^-- SC2086: Double quote to prevent globbing and word splitting.
                                 ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 68:
	for i in in $(find $2 -regex  '.*\.*'); do
                           ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 70:
		filename=$(basename $i)
                                    ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 73:
		cp $i $2/$name$1.$extension
                   ^-- SC2086: Double quote to prevent globbing and word splitting.
                      ^-- SC2086: Double quote to prevent globbing and word splitting.
                         ^-- SC2086: Double quote to prevent globbing and word splitting.
                              ^-- SC2086: Double quote to prevent globbing and word splitting.
                                 ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 81:
	for i in $(find "$1" -regex '.*\.svg\|.*\.png');do
                 ^-- SC2044: For loops over find output are fragile. Use find -exec or a while read loop.


In ../2015-linux-public-icoy/code/work_1.sh line 83:
		filename=$(basename $i)
                                    ^-- SC2086: Double quote to prevent globbing and word splitting.


In ../2015-linux-public-icoy/code/work_1.sh line 87:
		convert $i $1/$name.jpg
                        ^-- SC2086: Double quote to prevent globbing and word splitting.
                           ^-- SC2086: Double quote to prevent globbing and word splitting.
                              ^-- SC2086: Double quote to prevent globbing and word splitting.